src/kboot.c

```diff
@@ -286,6 +286,52 @@ static int dt_set_mac_addresses(void)
     return 0;
 }
 
+/* adjust PCI port properties, Required for adjusting max-link-speed
+ * for the 10 gigabit ehternet Mac Mini variant.
+ */
+static int dt_adjust_pcie_port_properties(void)
+{
+    int anode = adt_path_offset(adt, "/arm-io/apcie");
+
+    if (anode < 0)
+        bail("ADT: /arm-io/apcie not found\n");
+
+    ADT_FOREACH_CHILD(adt, anode)
+    {
+        const char *anode_name = adt_get_name(adt, anode);
+        if (strncmp(anode_name, "pci-bridge", 10) || strlen(anode_name) != 11)
+            continue;
+
+        char port_num = anode_name[10];
+        if (port_num < '0' || port_num > '9')
+            continue;
+
+        u32 max_link_speed;
+        if (ADT_GETPROP(adt, anode, "maximum-link-speed", &max_link_speed) < 0)
+            continue;
+
+        if (max_link_speed < 1 || max_link_speed > 16) {
+            printf("ADT: unexpected maximum-link-speed %u for %s\n", max_link_speed, anode_name);
+            continue;
+        }
+
+        char port_name[] = "port0_";
+        port_name[5] = port_num;
+
+        const char *path = fdt_get_alias(dt, port_name);
```

Comment:
How does this work? We don't actually have aliases (in `/aliases`) for the PCIe ports, as far as I know.